sdc/str_arr_ext.py

```diff
@@ -671,12 +671,16 @@ def construct_string_array(context, builder):
 @lower_builtin(StringArray, types.UniTuple)
 @lower_builtin(StringArray, types.Tuple)
 def impl_string_array_single(context, builder, sig, args):
-    if isinstance(args[0], types.UniTuple):
-        assert args[0].dtype == string_type
 
-    if isinstance(args[0], types.Tuple):
-        for i in args[0]:
-            assert i.dtype == string_type or i.dtype == types.StringLiteral
+    arg = args[0]
+    if isinstance(arg, (types.UniTuple, types.List)):
+        assert (arg.dtype == string_type
+                or (isinstance(arg, types.optional) and arg.type == string_type))
+
+    # TODO: check whether below works at all, otherwise it might be removed
+#     if isinstance(arg, types.Tuple):
+#         for i in arg:
+#             assert i.dtype == string_type or i.dtype == types.StringLiteral
```

Comment:
This should work. But shouldn't we also check for None type? Or optional... I'm not sure

Reply:
@AlexanderKalistratov No, actually I double-checked and it doesn't:
```
Invalid use of Function(<built-in function getitem>) with argument(s) of type(s): (Tupl
e(unicode_type, unicode_type, none, unicode_type, none, unicode_type, unicode_type), in
t64)
```
as tuple aggregates different types in this case, and applying getitem in a loop fail at:
```
    def str_arr_from_sequence(in_list):
        <source elided>
        for i in numba.prange(n_strs):
            s = in_list[i]
```
So this func only works for a UniTuple of unicode strings case now. I'll add a note about this.